### PR TITLE
Path.file name.to string

### PR DESCRIPTION
**Pull request by Tregz**
----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/main/kotlin/ch/tutteli/atrium/domain/robstoll/lib/creating/pathAssertions.kt [hunk 1]
```diff
@@ -23,6 +23,9 @@ fun <T : Path> _exists(assertionContainer: Expect<T>): Assertion =
 fun <T : Path> _existsNot(assertionContainer: Expect<T>): Assertion =
     ExpectImpl.builder.createDescriptive(assertionContainer, NOT_TO, RawString.create(EXIST)) { it.notExists }
 
+fun <T : Path> _fileName(assertionContainer: Expect<T>): ExtractedFeaturePostStep<T, String> =
+    ExpectImpl.feature.manualFeature(assertionContainer, DescriptionPathAssertion.FILE_NAME) { fileName.toString() }
```
Comment (robstoll) [inline domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/main/kotlin/ch/tutteli/atrium/domain/robstoll/lib/creating/pathAssertions.kt]:
```suggestion
    ExpectImpl.feature.manualFeature(assertionContainer, FILE_NAME) { fileName.toString() }
```

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt [hunk 2]
```diff
@@ -16,25 +16,27 @@ import java.nio.file.Paths
 abstract class PathFeatureAssertionsSpec(
     parentFeature: Feature0<Path, Path>,
     parent: Fun1<Path, Expect<Path>.() -> Unit>,
+    fileName: Feature0<Path, String>,
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt]: please rename to `fileNameFeature` (see for instance fileNameWithoutExtensionFeature)
and add `fileName : Fun1<....>` (see fileNameWithoutExtension)

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt [hunk 3]
```diff
@@ -100,6 +102,22 @@ abstract class PathFeatureAssertionsSpec(
...
+
+        context("File with extension") {
+            it("toBe(my) holds") {
+                expect(Paths.get("a/my.txt")).fileNameVal().toBe("my.txt")
+            }
+            it("toBe(my.txt) fails") {
+                expect {
+                    expect(Paths.get("a/my.txt")).fileNameVal().toBe("a/my.txt")
+                }.toThrow<AssertionError> {
+                    messageContains("$fileNameDescr: \"my\"")
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt]: I would expect something different (I hope this test fails)

**Comment by robstoll** [comment]
@Tregz please rebase on master and solve conflicts

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt [hunk 4]
```diff
@@ -16,25 +16,29 @@ import java.nio.file.Paths
...
 
-    val tempFolder = TempFolder.perTest()
+    val tempFolder = TempFolder.perTest() //or perGroup()
     registerListener(tempFolder)
 
     include(object : SubjectLessSpec<Path>(describePrefix,
         parentFeature.forSubjectLess().adjustName { "$it feature" },
-        parent.forSubjectLess() { },
+        parent.forSubjectLess { },
+        fileNameFeature.forSubjectLess(),
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt]:
```suggestion
        fileNameFeature.forSubjectLess().adjustName { "$it feature" },
```

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt [hunk 5]
```diff
@@ -174,7 +175,7 @@ abstract class PathAssertionsSpec(
                     expect(Paths.get("/path/ends/with/this"))
                         .endsNotWithFun(Paths.get("with/this"))
                 }.toThrow<AssertionError> {
-                    messageContains("${DescriptionPathAssertion.ENDS_NOT_WITH.getDefault()}:")
+                    messageContains("${DescriptionCharSequenceAssertion.ENDS_NOT_WITH.getDefault()}:")
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt]: Please use DescriptionPathAssertion, I guess this was a mistake

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt [hunk 6]
```diff
@@ -6,6 +6,7 @@ import ch.tutteli.atrium.api.verbs.internal.expect
 import ch.tutteli.atrium.specs.*
 import ch.tutteli.atrium.translations.DescriptionBasic.NOT_TO
 import ch.tutteli.atrium.translations.DescriptionBasic.TO
+import ch.tutteli.atrium.translations.DescriptionCharSequenceAssertion
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt]: remove (reason see next comment)

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt [hunk 7]
```diff
@@ -100,6 +104,52 @@ abstract class PathFeatureAssertionsSpec(
...
     }
 
+    describeFun("val ${fileNameFeature.name}") {
+        val fileNameVal = fileNameFeature.lambda
+
+        context("File with extension") {
+            it("toBe(my.txt) holds") {
+                expect(Paths.get("a/my.txt")).fileNameVal().toBe("my.txt")
+            }
+            it("toBe(my.txt) fails") {
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt]: This test case does not make sense in this context `"File with extension"` -> your case does not have an extension.
However, I suggest you rename the context to `context("path a/my.txt")` as fileName does not have anything to do with *does it have an extension or not?*

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt [hunk 8]
```diff
@@ -100,6 +104,52 @@ abstract class PathFeatureAssertionsSpec(
...
+                    messageContains("$fileNameDescr: \"my\"")
+                }
+            }
+        }
+    }
+
+    describeFun("fun ${fileName.name}") {
+        val fileNameFun = fileName.lambda
+
+        context("File with extension") {
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt]: same here, rename the context
```suggestion
        context("path a/my.txt") {
```

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt [hunk 9]
```diff
@@ -100,6 +104,52 @@ abstract class PathFeatureAssertionsSpec(
...
+            it("toBe(my.txt) fails") {
+                expect {
+                    expect(Paths.get("a/my")).fileNameFun { toBe("my.txt") }
+                }.toThrow<AssertionError> {
+                    messageContains("$fileNameDescr: \"my\"")
+                }
+            }
+        }
+
+        context("path with double extension") {
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathFeatureAssertionsSpec.kt]: IMO you can remove this context, this is not a special case your for function (is one for fileNameWithoutExtension)

File: translations/en_GB/atrium-translations-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/translations/DescriptionPathAssertion.kt [hunk 10]
```diff
@@ -3,9 +3,9 @@ package ch.tutteli.atrium.translations
 import ch.tutteli.atrium.reporting.translating.StringBasedTranslatable
 
 enum class DescriptionPathAssertion(override val value: String) : StringBasedTranslatable {
-    DOES_NOT_HAVE_PARENT("does not have a parent"),
-    ENDS_NOT_WITH("ends not with"),
     EXIST("exist"),
+    FILE_NAME("file name"),
```
Comment (robstoll) [inline translations/en_GB/atrium-translations-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/translations/DescriptionPathAssertion.kt]: Please keep the order of this class and the one of de_CH in sync

File: README.md [hunk 11]
```diff
@@ -1535,7 +1535,7 @@ But Atrium shows where it goes wrong and even gives a possible hint:
 expect: 9.99        (kotlin.Float <1234789>)
 ◆ to be (error ± 0.01): 10.0        (kotlin.Float <1234789>)
     » failure might be due to using kotlin.Float, see exact check on the next line
-    » exact check is |9.989999771118164 - 10.0| = 0.010000228881835938 ≤ 0.009999999776482582
+    » exact check is |9,989999771118164 - 10,0| = 0,010000228881835938 ≤ 0,009999999776482582
```
Comment (robstoll) [inline README.md]: please revert this change, was it intentional?

**Comment by robstoll** [comment]
@Tregz I did the revert myself. Thanks for your contribution :+1:

File: README.md (hunk 11 above)
Comment (Tregz) [reply README.md]: It was edited automatically. I do not know why? Could be a language formatting.

Comment (robstoll) [reply README.md]: Ah... I see, makes sense, your local number formatting is different that's why and the example in the README gets automatically generated based on some specs.
Fancy to fix the problem so that someone else with a different number formatting does not run into the same problem?

Comment (Tregz) [reply README.md]: I do not know about automatically generated text in the README. I had a look a #211 and went tru the ReadmeSpec. I don't understand well how it is done. I would need to spend required time to dig it. Sadly, I am too busy right now. If I had the time, it would have been great for me to learn how to do it, like for this contribution.

Comment (robstoll) [reply README.md]: No worries, and you are always welcome to contribute again if you have time (also after hacktoberfest :grin: )
